src/posix/client.c

```diff
@@ -26,7 +26,7 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "openthread-core-config.h"
+#include "platform/platform-config.h"
```

Comment:
Should we be renaming this include to something like `openthread-posix-config.h`?

Reply:
Good idea. 
Renamed it to `openthread-posix-config.h` (also updated the first commit message) in the new push.